### DISCUSSION TITLE
What interface am I affecting when setting <child>.interface?

### DISCUSSION THREAD

**Discussion by Kentzo** [strongswan/strongswan#1723] [Q&A] [answered]
This swanctl option is documented as:

> Since version 5.5.0. Optional interface name to restrict outbound IPsec policies

I'm configuring a responder. What is considered an outbound policy in this case and how is it different from `interfaces_use`? Is it just to narrow down the list of interfaces the initiator can be accepted from for a given connection?

**Comment by tobiasbrunner** [comment]
What's unclear about the term "outbound **IPsec** policy"? Also note the location of the option in **child** sections.

> What is considered an outbound policy in this case and how is it different from `interfaces_use`? Is it just to narrow down the list of interfaces the initiator can be accepted from for a given connection?

The **global** `interface_use` option has obviously nothing to do with the other. It controls on which interfaces IKE packets are accepted.

**Reply by Kentzo** [reply]
Does it restrict local interfaces that can be considered for outbound (responder -> initiator) policies?

**Reply by Kentzo** [reply]
After a bit of research I learned that this parameter is used to set interface of XFRM policies (`dev` of SELECTOR in ip-xfrm). By default strongSwan uses the interface that accepted the connection.

Out of curiosity, when it might be useful to set it to something else?

**Reply by tobiasbrunner** [reply, marked as answer]
> By default strongSwan uses the interface that accepted the connection.

No, it doesn't set one at all (i.e. outbound/forwarded packets with matching addresses are processed on any interface).

> Out of curiosity, when it might be useful to set it to something else?

It allows restricting the policy to a single interface, so I guess there could be some use cases (one is offloading bypass policies to hardware that supports it). And e.g. the [ikev2/shunt-manual-prio example](https://strongswan.org/testing/testresults/ikev2/shunt-manual-prio/) installs a drop policy only on the external interface.

**Reply by Kentzo** [reply]
I need to look deeper into how "shunting" via XFRM policies work in strongSwan.
